sdk/core/azure-core/src/datetime.cpp

```diff
@@ -430,12 +437,12 @@ DateTime DateTime::FromString(std::string const& dateString, DateFormat format)
     }
     else
     {
-      return result;
+      throw DateTimeException("Error parsing DateTime: Day of the month.");
     }
 
     if (monthDay == 0)
     {
-      return result;
+      throw DateTimeException("Error parsing DateTime: Invalid month day number (0).");
```

Comment:
Missing argument for `{0}`?

Reply:
No, it literally says - "the number is invalid (ant that number is zero) :)"